airflow/models/__init__.py

```diff
@@ -359,6 +356,9 @@ def process_file(self, filepath, only_if_updated=True, safe_mode=True):
         if filepath is None or not os.path.isfile(filepath):
             return found_dags
 
+        if filepath.endswith(".pyc"):
+            filepath = filepath.rstrip("c")
```

Comment:
.pyc files should be ignored, and not here, but earlier. We should never get to this function with a dag.pyc file.

Reply:
I do agree but in testing I sometimes do see this happening. I think this is happening because there are some inline DAG's inside the tests that are only part of a method and not a separated file. This pyc file will when end up in DagModel.fileloc.

I can try to raise an exception to make a list of tests where this is happening if you want?

Reply:
Please do, this shouldn't happen. While listing the directory, the `.pyc` should be ignored.

Reply:
Will remove this, if the testing are succeeded this should be fine

Reply:
Seems the testing is still successful, resolving this now